raiden/transfer/views.py

```diff
@@ -315,6 +315,43 @@ def get_channelstate_for(
     return channel_state
 
 
+def get_first_channelstate(
+    chain_state: ChainState,
+    payment_network_id: PaymentNetworkID,
+    token_address: TokenAddress,
+    creator_address: Address = None,
+) -> Optional[NettingChannelState]:
+    """ Return the NettingChannelState if it exists, None otherwise. """
+    token_network = get_token_network_by_token_address(
+        chain_state, payment_network_id, token_address
+    )
+
+    channel_state = None
+    address_to_get_channel_state = creator_address
+
+    channel = None
+    if token_network and creator_address in token_network.channelidentifiers_to_channels:
+        channels = []
+
+        for channel_id in token_network.partneraddresses_to_channelidentifiers[creator_address]:
+
+            if creator_address in token_network.channelidentifiers_to_channels:
+                channel = token_network.channelidentifiers_to_channels[creator_address].get(channel_id)
+
+            if channel is not None:
+                if channel.close_transaction is None or channel.close_transaction.result != 'success':
```

Comment:
Why a nested ¡if¡ instead of an 'and'?

Reply:
I could sugges to encapsulate all this on a method or a simple function that does all these checks instead, nesting if's looks weird

Reply:
unused function, removing it